pyomo/contrib/alternative_solutions/balas.py

```diff
@@ -0,0 +1,246 @@
+import pyomo.environ as pe
+from pyomo.common.collections import ComponentSet
+from pyomo.contrib.alternative_solutions import Solution
+import pyomo.contrib.alternative_solutions.aos_utils as aos_utils
+
+
+def enumerate_binary_solutions(
+    model,
+    *,
```

Comment:
I don't have strong feelings about this - the question mainly came up because of user experience. If you provide defaults, then the user does not have to put in everything.

Reply:
That's still true with this syntax.  With defaults, you don't need to specify the arguments after the "*".  But any arguments you do specify must be with a keyword.  For example

`enumerate_binary_solutions(model, quiet=True)`